configure.ac

```
@@ -1185,7 +1185,11 @@ SQUID_CHECK_LIB_WORKS(ldap,[
     dnl On MinGW OpenLDAP is not available, try Windows LDAP libraries
     dnl TODO: use AC_CHECK_LIB
     LIBLDAP_LIBS="-lwldap32"
-    AC_CHECK_HEADERS(winldap.h)
+    AC_CHECK_HEADERS(winldap.h,,,[
```

I cannot tell for sure that windows.h is _always_ checked for before this code. AFAICT, Squid code adds windows.h to avoid such order dependency problems in some cases (e.g., in squid_host_os and enable_eui blobs):

```suggestion
    AC_CHECK_HEADERS( \
        windows.h \
        winldap.h,,,[
```

The above suggestion is untested, may need adjustments, and I do not insist on it.

---

Agreed, though the change should be this:
```suggestion
    AC_CHECK_HEADERS([windows.h winldap.h],,,[
```

---

I can add it as you both agree this can be done.
windows.h is tested on line 207 of configure.ac
```
AS_CASE(["$squid_host_os"],
  [mingw],[
    CFLAGS="$CFLAGS -mthreads"
    CXXFLAGS="$CXXFLAGS -mthreads -static-libgcc -static-libstdc++"
    dnl Check for Winsock only on MinGW
    SQUID_CHECK_WINSOCK_LIB
    AC_CHECK_HEADERS([ws2tcpip.h winsock2.h windows.h io.h],,,[
```

So it is not unconditional, but it is done in case of host os being mingw.
In fact, in my run testing, windows.h gets checked at line 187 of the output, then again at line 225, and winldap.h at line 251.
But testing a third time won't hurt much due to caching, so sure